packages/components/src/entities/SampleAliquotsGridPanel.tsx

```diff
@@ -114,98 +117,116 @@ interface Props {
     lineageUpdateAllowed: boolean;
     onSampleChangeInvalidate: (schemaQuery: SchemaQuery) => void;
     user: User;
+    showLabelOption?: boolean
 }
 
-interface State {
-    showConfirmDelete: boolean;
-}
+export const SampleAliquotsGridPanelImpl: FC<Props & InjectedQueryModels> = memo(props => {
+    const { actions, queryModels, showLabelOption = true, ...buttonProps } = props;
+    const [showConfirmDelete, setConfirmDelete] = useState<boolean>(false);
+    const [showPrintDialog, setShowPrintDialog] = useState<boolean>(false);
+    const { createNotification } = useNotificationsContext();
+    const { canPrintLabels, printServiceUrl, labelTemplate } = useLabelPrintingContext();
 
-export class SampleAliquotsGridPanelImpl extends PureComponent<Props & InjectedQueryModels, State> {
-    state: Readonly<State> = { showConfirmDelete: false };
+    const queryModel = queryModels[MODEL_ID];
 
-    getQueryModel = (): QueryModel => {
-        return Object.values(this.props.queryModels)[0];
-    };
+    const resetState = useCallback((): void => {
+        setConfirmDelete(false);
+        setShowPrintDialog(false);
+    },[]);
 
-    afterAction = (): void => {
-        const { actions, onSampleChangeInvalidate } = this.props;
-        const model = this.getQueryModel();
+    const afterAction = useCallback((): void => {
+        const { onSampleChangeInvalidate } = props;
 
-        this.resetState();
-        onSampleChangeInvalidate(model.schemaQuery);
-        actions.loadModel(model.id, true);
-    };
+        resetState();
+        onSampleChangeInvalidate(queryModel.schemaQuery);
+        actions.loadModel(queryModel.id, true);
+    }, [actions, queryModel]);
```

Comment:
looks like IntelliJ is saying there are some missing deps here for the `afterAction`